modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/TypedInput.es.js

```diff
@@ -42,13 +40,98 @@ class TypedInput extends React.Component {
 				selected={value}
 			/>;
 	}
+	_handleDateChange = dateObject => {
+		const {onChange} = this.props;
+		const newValue = dateObject.toISOString();
+
+		onChange(newValue, PROPERTY_TYPES.DATE);
+	}
+	_renderDateType = () => {
+		const {value} = this.props;
+
+		return (<div className="criterion-input">
+			<DatePicker
+				className="form-control"
+				onChange={this._handleDateChange}
+				selected={new Date(value)}
+			/>
+		</div>);
+	}
+	_renderBooleanType = () => {
+		const {value} = this.props;
+
+		return (
+			<ClaySelect
+				className="criterion-input form-control"
+				data-testid="options-boolean"
+				onChange={this._handleChange}
+				options={[
+					{
+						label: 'TRUE',
+						value: 'true'
+					},
+					{
+						label: 'FALSE',
+						value: 'false'
+					}
+				]}
+				selected={value}
+			/>);
+	}
+	_handleNumberChange = event => {
+		const {onChange} = this.props;
+		const value = parseInt(event.target.value, 10);
+
+		if (onChange && !isNaN(value)) {
+			onChange(value.toString());
+		}
+	}
+	_renderNumberType = () => {
```

Comment:
I know this is still under a work in progress, but just a friendly reminder to add new lines between the methods. 😄